src/utils/executeAsTask.ts

```diff
@@ -34,5 +34,15 @@ export async function executeAsTask(context: IActionContext, command: string, na
         task.definition.idRandomizer = Math.random();
     }
 
-    return vscode.tasks.executeTask(task);
+    const taskExecution = await vscode.tasks.executeTask(task);
+
+    const taskEndPromise = new Promise<void>((resolve) => {
+        vscode.tasks.onDidEndTaskProcess(e => {

```

Comment:
The event subscription returns a `Disposable` that should be invoked prior to resolving, else this may get continued to get called after the task is done.

Reply:
Good catch, thank you!